### ISSUE TITLE
Adding number and string should work?

### ISSUE THREAD

**Issue by tcard** [munificent/craftinginterpreters#41]
According to [3.4.1](http://www.craftinginterpreters.com/the-lox-language.html#arithmetic):

> If either operand of + is a string, then the other operand is converted to a string (if it isn’t already one) and the results are concatenated.

But both interpreter currently say:

> Operands must be two numbers or two strings.

**Comment by egordorichev** [comment]
https://github.com/munificent/craftinginterpreters/pull/42

**Comment by munificent** [comment]
Oof, you're right! I wrote that chapter quite a while after implementing that behavior in Lox and I'm embarrassed to admit I misremembered how my own language works!

I think the simpler fix is to update the text to describe how Lox actually behaves. In a real language, it's probably more usable to convert either operand to a string, but doing that in Lox would add a decent amount of fairly boring code, so the simplest thing is to just leave it out and leave "+" a little less flexible.

I'll fix the prose.
